### PR TITLE
Fixes #94 - Changes to support Android Chrome

### LINKED ISSUE #94: KeyboardEvent.key not working on certain Android devices
While testing for issue #84 on Android I ran into an issue, whereby the KeyBoardEvent.key was producing undefined, causing issues with matching against delimiterChars. At the same time the default delimiters aren't useable on phone keyboard, due to lack of tab key and enter not being available, other than when suggested tags are shown.

When connected to a computer, for debugging the view in Chrome on the computer does provide event.key values on the phone, but not when typing from the phone's keyboard.

A limitation with `event.keyCode`, if we used that as fallback, is that 'A', 'a' and '@' all provide the same value, when using the US English layout.

I tried using [keyboardevent-key-polyfill](https://www.npmjs.com/package/keyboardevent-key-polyfill), but that didn't prove of use since 'event.key' is technically supported, but not receiving the right values from the on-screen keyboard.

My initial tests were with an LG Stylo 2, running Android 7 and Chrome 60, though I did test with two other Android phones and got the same behaviour.

Note, that I did not experience any issue when testing with iOS.

Edit: Have opened a ticket against Chromium, to see if this could confirm a device bug or provide a lead to the issue cause:

https://bugs.chromium.org/p/chromium/issues/detail?id=763559

Update: per the response on the referenced ticket we should look into using `oninput` or `onchange` instead, since the limitation encountered is apparently due to the behaviour of the 'soft keyboards'. 

Update: oninput won't provide us the values of non-printing characters. onchange only gives us the the new state. The ticket was marked as "Won't fix" which is a little disconcerting, since they have essentially broken `onKeyDown` when it comes to non-deprecated values.

### PR DESCRIPTION
**Pull request by ajmas**
Changes to ensure we support Google Chrome on Android, due to limitations around the soft keyboards and `KeyboardEvent.key` in `onKeydown()`.

Note, current PR's code coverage is reduced, but it is more to get feedback at this point.

### CODE REVIEW

**Comment by ajmas** [comment]
Note, there were two three cases where there was no reasonable way of creating proper code coverage, so I opted to exclude them. Let me know if you have any issues with that.

File: README.md [hunk 1]
```diff
@@ -121,11 +121,11 @@ Boolean parameter to control whether the text-input should be automatically resi
 
 #### delimiters (optional)
 
-Array of integers matching keyboard event `keyCode` values. When a corresponding key is pressed, the preceding string is finalised as tag. Default: `[9, 13]` (Tab and return keys).
+Array of integers matching keyboard event `keyCode` values. When a corresponding key is pressed, the preceding string is finalised as tag. Best used for non-printable keys, such as the tab and enter/return keys. Default: `[9, 13]` (Tab and return keys).
 
 #### delimiterChars (optional)
 
-Array of characters matching keyboard event `key` values. This is useful when needing to support a specific character irrespective of the keyboard layout. Note, that this list is separate from the one specified by the `delimiters` option, so you'll need to set the value there to `[]`, if you wish to disable those keys. Example usage: `delimiterChars={[',', ' ']}`. Default: `[]`
+Array of characters matching characters that can be displayed in an input field. This is useful when needing to support a specific character irrespective of the keyboard layout, such as for internationalisation. Example usage: `delimiterChars={[',', ' ']}`. Default: `[',']`
```
Comment (i-like-robots) [inline README.md]: Nice distinction 👍

File: README.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ Override the default class names. Defaults:
 
 #### handleAddition (required)
 
-Function called when the user wants to add a tag. Receives the tag.
+Function called when the user wants to add one or more tags. Receives the tag or tags. Value can be a tag or an Array of tags.
```
Comment (i-like-robots) [inline README.md]: I think this could be considered a breaking change, it might better to invoke the callback once for each tag to avoid that?

File: lib/ReactTags.js [hunk 3]
```diff
@@ -40,28 +40,124 @@ class ReactTags extends React.Component {
...
+   * Handles the value changes to the input field and uses the `delimiterChars`
+   * property to know on what character to try to create a tag for. Only characters
+   * valid for display in an `input` field are supported. Other values passed,
+   * such as 'Tab' and 'Enter' cause adverse effects.
+   *
+   * Note, this method is necessary on Android, due to a limitation of the
+   * `KeyboardEvent.key` having an undefined value, when using soft keyboards.
+   * in Android's version of Google Chrome. This method also handles the paste
+   * scenario, without needing to provide a supplemental 'onPaste' handl+er.
+   */
```
Comment (i-like-robots) [inline lib/ReactTags.js]: Thanks for the description!

**Comment by i-like-robots** [comment]
@ajmas wow, good work! This (to me) is quite a bit nicer than the onPaste implementation, and great work covering the edge cases too. I will give this a good test today and let's get it sorted this week as it's dragged on for too long (my fault entirely).

**Comment by i-like-robots** [comment]
Re. the test coverage @ajmas - I really don't mind, so long as the main features and interface is covered then I'm not bothered about implementation detail being missed.

File: README.md (hunk 2 above)
Comment (ajmas) [reply README.md]: It is, but I couldn't seem to get things to behave otherwise, when I was testing with the example. I can make another branch in my fork to show you the issue?
